draft-ietf-tls-tls13.md

```diff
@@ -2720,57 +2705,30 @@ The "extension_data" field of this extension contains an
        struct {
            select (Role) {
                case client:
-                   opaque configuration_id<1..2^16-1>;
-                   CipherSuite cipher_suite;
-                   Extension extensions<0..2^16-1>;
                    opaque context<0..255>;
 
                case server:
                   struct {};
            }
        } EarlyDataIndication;
 
-configuration_id
-: The label for the configuration in question.
-
-cipher_suite
-: The cipher suite which the client is using to encrypt the early data.
-
-extensions
-: The extensions required to define the cryptographic configuration
-  for the clients early data (see below for details).
-
 context
 : An optional context value that can be used for anti-replay
   (see below).
 {:br }
 
-The client specifies the cryptographic configuration for the 0-RTT
-data using the "configuration_id", "cipher_suite", and "extensions"
-values. For configurations received in-band (in a previous TLS connection)
-the client MUST:
-
-- Send the same cryptographic determining parameters (Section
-{{cryptographic-determining-parameters}}) with the previous
-connection. If a 0-RTT handshake is being used with a PSK
-that was negotiated via a non-PSK handshake,
-then the client MUST use the same symmetric cipher parameters
-as were negotiated on that handshake but with a PSK cipher
-suite.
-
-- Indicate the same parameters as the server indicated in that connection.
+All of the parameters for the 0-RTT data (symmetric cipher suite,
+ALPN, etc.) MUST be those which were negotiated in the connection
+which established the PSK.  The PSK used to encrypt the early data
+MUST be the first PSK listed in the client's "pre_shared_key" extension.
 
 0-RTT messages sent in the first flight have the same content types
 as their corresponding messages sent in other flights (handshake,
 application_data, and alert respectively) but are protected under
 different keys. After all the 0-RTT application data messages (if
 any) have been sent, a "end_of_early_data" alert of type
 "warning" is sent to indicate the end of the flight.
-Clients which do 0-RTT MUST always send "end_of_early_data" even
-if the ServerConfiguration indicates that no application
-data is allowed (EarlyDataType of "client_authentication"),
-though in that case it MUST NOT send any non-empty data records
-(i.e., those which consist of anything other than padding).
+Clients which do 0-RTT MUST always send "end_of_early_data".
```

Comment:
This could be read as having to send the alert on every connection.

Maybe "0-RTT data MUST be followed by an "end_of_early_data" alert."